map/place_data.hpp

```diff
@@ -0,0 +1,163 @@
+  osm::YesNoUnknown IsOpen() const { return m_isOpenNow; }
+  bool IsSponsoredHotel() const { return m_isSponsoredHotel; }
+
+private:
+  void ConfigSubtitle(search::Metadata const & m)
+  {
+    vector<string> values;
+    // Type
+    values.push_back(m_owner->GetFeatureType());
+    bool const isSearch = m.m_isInitialized;
+
+    // Cuisine
+    for (auto const & cuisine : GetLocalizedCuisines(isSearch ? m.m_cuisine :
+                                                     m_metadata->Get(feature::Metadata::FMD_CUISINE)))
+      values.push_back(cuisine);
+
+    // Hotel's stars
+    auto const stars = FormatStars(isSearch ? m.m_stars : m_metadata->Get(feature::Metadata::FMD_STARS));
+    if (!stars.empty())
+      values.push_back(stars);
+
+    // Operator
+    auto const op = isSearch ? m.m_operator : m_metadata->Get(feature::Metadata::FMD_OPERATOR);
+    if (!op.empty())
+      values.push_back(op);
+
+    // Elevation
+    auto const ele = GetElevationFormatted(isSearch ? m.m_elevation : m_metadata->Get(feature::Metadata::FMD_ELE));
+    if (!ele.empty())
+      values.push_back(kMountainSymbol + ele);
+
+    m_subtitle = strings::JoinStrings(values, kSubtitleSeparator);
+  }
+
+  vector<string> GetLocalizedCuisines(string const & cuisineStr) const
+  {
+    vector<string> localized;
+    osm::Cuisines::Instance().ParseAndLocalize(cuisineStr, localized);
+    return localized;
+  }
+
+  string FormatStars(string const & starsStr) const
+  {
+    string stars;
+    int count = 0;
+    strings::to_int(starsStr, count);
+    for (int i = 0; i < count; ++i)
+      stars.append(kStarSymbol);
+
+    return stars;
+  }
+
+  string GetElevationFormatted(string const & elevationStr) const
+  {
+    double value;
+    if (strings::to_double(elevationStr, value))
+      return measurement_utils::FormatAltitude(value);
+
+    LOG(LWARNING, ("Invalid metadata for elevation:", elevationStr));
+    return {};
+  }
+
+  void ConfigHotelRating(string const & ratingStr)
+  {
+    if (!IsSponsoredHotel())
+      return;
+
+    m_hotelRating = ratingStr.empty() ? kEmptyRatingSymbol : ratingStr;
+  }
+
+  void ConfigApproximatePricing(string const & pricingStr)
+  {
+    if (!IsSponsoredHotel())
+      return;
+
+    int pricing;
+    strings::to_int(pricingStr, pricing);
+    string result;
+    for (auto i = 0; i < pricing; i++)
+      result.append(kPricingSymbol);
+    
+    m_approximatePricing = result;
+  }
+
+  void ConfigIsOpen(string const & openHoursStr)
+  {
+    if (openHoursStr.empty())
+      return;
+    osmoh::OpeningHours const oh(openHoursStr);
+    // TODO: We should check closed/open time for specific feature's timezone.
+    time_t const now = time(nullptr);
+    if (oh.IsValid() && !oh.IsUnknown(now))
+      m_isOpenNow = oh.IsOpen(now) ? osm::Yes : osm::No;
+    // In else case value us osm::Unknown, it's set in preview's constructor.
+  }
+
+private:
+  feature::Metadata const * m_metadata;
+  Owner const * m_owner;
```

Comment:
I guess I can't store const reference because sometimes Data is constructed by its default constructor.